ilastik/applets/dataSelection/opDataSelection.py

```diff
@@ -19,195 +19,353 @@
+            self.original_paths = splitPath(filepath)
+            self.expanded_paths = self.expand_path(filepath, cwd=self.base_dir)
+            assert len(self.expanded_paths) == 1 or self.sequenceAxis
+            if len({PathComponents(ep).extension for ep in self.expanded_paths}) > 1:
+                raise Exception(f"Multiple extensions unsupported as a single data source: {self.expanded_paths}")
+            self.nickname = nickname or self.create_nickname(self.expanded_paths)
+            self.filePath = os.path.pathsep.join(self.expanded_paths)
+            op_reader = OpInputDataReader(graph=Graph(),
+                                          WorkingDirectory=self.base_dir,
+                                          FilePath=self.filePath,
+                                          SequenceAxis=self.sequenceAxis)
+            meta = op_reader.Output.meta
+            default_tags = meta.axistags
+            self.laneShape = meta.shape
+            self.laneDtype = meta.dtype
+            self.drange = drange or meta.get('drange')
+        else: # path is url
             self.filePath = filepath
-            self.fromstack = fromstack
-            self.sequenceAxis = sequence_axis
+            self.expanded_paths = [filepath]
+
+        if isinstance(self.laneDtype, numpy.dtype):
+            self.laneDtype = numpy.typeDict[self.laneDtype.name]
 
         if jsonNamespace is not None:
             self.updateFromJson(jsonNamespace)
+        self.normalizeDisplay = (self.drange is not None) if normalizeDisplay is None else normalizeDisplay
+
+        self.axistags = axistags or default_tags
+        if len(self.axistags) != len(self.laneShape):
+            if not fill_in_dummy_axes:
+                raise Exception("Axistags {self.axistags} don't fit data shape {self.laneShape}")
+            default_keys = [tag.key for tag in default_tags]
+            tagged_shape = {k: v for k, v in zip(default_keys, self.laneShape)}
+            squeezed_shape = {k: v for k, v in tagged_shape.items() if v != 1}
+            requested_keys = [tag.key for tag in axistags]
+            if set(requested_keys).issubset(set(default_keys)) and set(default_keys) - set(requested_keys) == set('c'):
+                self.axistags = default_tags #allow missing 'c' in axistags; not sure if this is a good idea
+            elif len(requested_keys) == len(squeezed_shape):
+                dummy_axes = [key for key in 'ctzxy' if key not in requested_keys]
+                out_axes = ""
+                for k, v in tagged_shape.items():
+                    if v > 1:
+                        out_axes += requested_keys.pop(0)
+                    else:
+                        out_axes += dummy_axes.pop(0)
+                self.axistags = vigra.defaultAxistags(out_axes)
+            else:
+                raise Exception(f"Cannot reinterpret input with shape {self.laneShape} using aksiskeys {requested_keys}")
+
+        if self.location is None:
```

Comment:
There's now a new class, `RelativeFilesysteDatasetInfo` which has to be used if one wants to store relative paths. Otherwise, its parent, `FilesystemDatasetInfo`, will just remember the absolute paths